### PR TITLE
Update glossary

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

* updated Glossary with IOTA 2.0 terms
* fixed import in glossary.mdx to use our swizzled component
* made sorting in Glossary component case-insensitive

## Links to any relevant issues

fixes https://github.com/iotaledger/devx/issues/300
fixes https://github.com/iotaledger/devx/issues/304

## Type of change

- Documentation Enhancement

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: docs/learn/glossary.mdx [hunk 1]
```diff
@@ -3,6 +3,6 @@ title: Glossary
 description: Glossary of all specialized names and phrases used in the IOTA and Shimmer networks. If you dont understand a wording you will find answers here.
 ---
 
-import Glossary from '@theme/Glossary';
+import Glossary from '/theme/src/theme/Glossary';
```
Comment (Dr-Electron) [inline docs/learn/glossary.mdx:6]: This change shouldn't be needed 🤔

Comment (Dr-Electron) [reply docs/learn/glossary.mdx:6]: Let me try :trollface: 

```suggestion
import Glossary from '@theme/Glossary';
```

**Comment by Dr-Electron** [comment]
Preview: https://iota-wiki-jtm8wa47o-iota1.vercel.app/

File: docs/get-started/sidebars.ts [hunk 2]
```diff
@@ -85,7 +85,11 @@ module.exports = {
       ],
     },
     'research-papers',
-    'glossary',
+    {
+      type: 'link',
+      href: '/learn/glossary',
```
Comment (Dr-Electron) [inline docs/get-started/sidebars.ts:90]: Do we want to link to glossary here?
I would either:
- Have its own page
- Remove the glossary in one of the sections

I guess I prefer the later

Comment (lucas-tortora) [reply docs/get-started/sidebars.ts:90]: Let's remove it then :)

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**